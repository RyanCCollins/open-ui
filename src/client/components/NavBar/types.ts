import * as React from 'react';
import NavBar from './';

interface Navlink { url: string; text: string; }
export interface NavBarProps extends React.Props<typeof NavBar> {
  logoText: string;
  links: Navlink[];
}
