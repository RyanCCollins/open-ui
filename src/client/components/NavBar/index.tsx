import * as React from 'react';
import { Anchor } from 'ui';
import { NavBarProps } from './types';
import {
  Nav,
  InnerNav,
  LeftMenu,
  RightMenu,
  Menu,
  MenuItem,
} from './styles';

export default function NavBar({
  logoText,
  links,
}: NavBarProps): JSX.Element {
  return (
    <Nav>
      <InnerNav>
        <LeftMenu>
          <Anchor path="/" color="#007acc">
            {logoText}
          </Anchor>
        </LeftMenu>
        <RightMenu>
          <Menu>
            {links.map((item, i) =>
              <MenuItem key={i}>
                <Anchor
                  path={item.url}
                >
                  {item.text}
                </Anchor>
              </MenuItem>,
            )}
          </Menu>
        </RightMenu>
      </InnerNav>
    </Nav>
  );
}
