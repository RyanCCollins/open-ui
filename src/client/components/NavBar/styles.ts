const styled = require('styled-components').default;

export const Nav = styled.nav`
  width: 100%;
  background: #0a0a0a;
  color: #fefefe;
  padding: .5rem;
  box-sizing: border-box;
`;

export const InnerNav = styled.div`
  width: 100%;
  display: flex;
  flex-direction: row;
  align-items: center;
  justify-content: space-between;
`;

export const LeftMenu = styled.div`
  margin-right: 1rem;
  margin-left: 1rem;
`;

export const RightMenu = styled.div`
  max-width: 400px;
  margin-right: 1rem;
`;

export const Menu = styled.ul`
  margin: 0;
  list-style-type: none;
`;

export const MenuItem = styled.li`
  display: table-cell;
  vertical-align: middle;
  a {
    color: #fff;
    display: block;
    padding: .7rem 1rem;
    line-height: 1;
  }
`;
