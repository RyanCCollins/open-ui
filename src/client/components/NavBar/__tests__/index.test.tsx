import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import * as React from 'react';
import { navbarProps } from './__mocks__/navBarMocks.mock';
import NavBar from '../';

describe('<NavBar /> default', () => {
  it('should render NavBar with links and text', () => {
    const wrapper = shallow(
      <NavBar {...navbarProps} />,
    );
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });
});
