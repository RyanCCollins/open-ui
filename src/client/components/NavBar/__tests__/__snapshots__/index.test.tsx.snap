// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NavBar /> default should render NavBar with links and text 1`] = `
<styled.nav>
  <styled.div>
    <styled.div>
      <withRouter(Anchor)
        color="#007acc"
        path="/"
      >
        NavBar Test
      </withRouter(Anchor)>
    </styled.div>
    <styled.div>
      <styled.ul>
        <styled.li>
          <withRouter(Anchor)
            path="https://medium.com"
          >
            Medium
          </withRouter(Anchor)>
        </styled.li>
        <styled.li>
          <withRouter(Anchor)
            path="https://www.ryancollins.io/"
          >
            RyanCollins
          </withRouter(Anchor)>
        </styled.li>
      </styled.ul>
    </styled.div>
  </styled.div>
</styled.nav>
`;
